internal/apis/certmanager/types_certificate.go

```diff
@@ -218,6 +234,14 @@ type CertificateSpec struct {
 	// issuer does not support CSRs with these X509 KeyUsage/ ExtKeyUsage extensions.
 	EncodeUsagesInRequest *bool
 
+	// Whether the BasicConstraints extension should be set in the encoded CSR.
+	//
+	// IMPORTANT: This option is only available if `UseCertificateRequestBasicConstraints` feature
+	// gate is set. Otherwise, the CSR will not have the BasicConstraints extension.
+	// This option defaults to true, and should only be disabled if the target
+	// issuer does not support CSRs with a X509 BasicConstraints extension.
+	EncodeBasicConstraintsInRequest *bool
```

Comment:
I try to be consistent with `EncodeUsagesInRequest`.

Reply:
My soft preference would be for us to add this only if we see a definite need for it - i.e. if we observe a breakage because of encoding the basicConstraints extension in the CSR. Without evidence of why we'd need this, it seems like clutter.